<launch>
  <arg name="gui" default="true"/>

  <include file="$(find jsk_pcl_ros_utils)/sample/include/play_rosbag_shelf_bin.xml"/>

  <node name="sample_manager"
        pkg="nodelet" type="nodelet"
        args="manager"/>

  <node name="pointcloud_to_cluster_point_indices"
        pkg="nodelet" type="nodelet"
        args="load jsk_pcl_utils/PointCloudToClusterPointIndices sample_manager">
    <remap from="~input" to="/right_hand_camera/depth_registered/points"/>
    <rosparam>
      skip_nan: true
    </rosparam>
  </node>

  <node name="cluster_point_indices_to_point_indices"
        pkg="nodelet" type="nodelet"
        args="load jsk_pcl_utils/ClusterPointIndicesToPointIndices sample_manager">
    <remap from="~input" to="pointcloud_to_cluster_point_indices/output"/>
    <rosparam>
      index: -1
    </rosparam>
  </node>

  <node name="point_indices_to_mask_image"
        pkg="nodelet" type="nodelet"
        args="load jsk_pcl_utils/PointIndicesToMaskImage sample_manager">
    <remap from="~input" to="cluster_point_indices_to_point_indices/output"/>
    <remap from="~input/image" to="/right_hand_camera/rgb/image_rect_color"/>
    <rosparam>
      approximate_sync: true
    </rosparam>
  </node>

  <group if="$(arg gui)">
    <node name="rviz"
          pkg="rviz" type="rviz"
          args="-d $(find jsk_pcl_ros_utils)/sample/config/sample_pointcloud_to_cluster_point_indices.rviz"/>
  </group>
</launch>
